### PR TITLE
refactor: make getorders method modular and generic

### LINKED ISSUE #18: docs: add swagger documentation
## Description
- Should add Swagger documentation to properly track the endpoints of the API and their respective descriptions.
## Features
- Add Swagger documentation.

### LINKED ISSUE #19: refactor: getorders method
## Description

- GetOrders method has too many lines of code. Making the method shorter using helper functions will make it easier to maintain and to read in the long run.
- GetOrders method is not generic. Making the method more generic will avoid duplicating code when adding more endpoints.

## Changes

- Make paginated search results more generic.
- Abstract logic into different helper functions.

### PR DESCRIPTION
**Pull request by NicolasRampoldi**
## Description

- Deleted order_by_constants file. It was unnecessary, this constants should be present in models.
- Created order_options model for the sort and direction constants.
- Created paginated_search_result model to abstract all the pagination properties in one object.
- Implemented addLinkHeader function to remove Link header's building from GetOrders method.

## Results
- This PR improves readability and abstraction of the GetOrders method.

closes #18 #22

### CODE REVIEW

**Comment by JuArce** [comment]
Please, use a more descriptive name for the PR

**Comment by NicolasRampoldi** [comment]
> Please, use a more descriptive name for the PR

✅

File: configs/swagger.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import (
+	docs "yab-explorer/docs"
+)
+
+func InitSwagger(apiPort string) {
+	docs.SwaggerInfo.Title = "YAB Explorer API"
+	docs.SwaggerInfo.Description = "YAB Explorer API"
+	docs.SwaggerInfo.Version = "1.0"
+	docs.SwaggerInfo.Host = "localhost:" + apiPort
```
Comment (JuArce) [inline configs/swagger.go]: Is there a way to add multiple hosts?
Ie here https://api.yetanotherswap.com/docs/ we have 3 servers available

File: docs/docs.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "type": "array",
+                            "items": {
+                                "$ref": "#/definitions/models.Order"
+                            }
+                        }
+                    },
+                    "400": {
```
Comment (JuArce) [inline docs/docs.go:64]: Add examples for error messages

**Review by JuArce — APPROVED** [review]

**Pull request merged**